pkg/config/v3alpha/config.go

```diff
@@ -0,0 +1,352 @@
+
+	// String fields
+	Domain     string `json:"domain,omitempty"`
+	Repository string `json:"repo,omitempty"`
+	Name       string `json:"projectName,omitempty"`
+	Layout     string `json:"layout,omitempty"`
+
+	// Boolean fields
+	MultiGroup      bool `json:"multigroup,omitempty"`
+	ComponentConfig bool `json:"componentConfig,omitempty"`
+
+	// Resources
+	Resources []resource.Resource `json:"resources,omitempty"`
+
+	// Plugins
+	Plugins PluginConfigs `json:"plugins,omitempty"`
+}
+
+// PluginConfigs holds a set of arbitrary plugin configuration objects mapped by plugin key.
+// TODO: do not export this once internalconfig has merged with config
+type PluginConfigs map[string]pluginConfig
+
+// pluginConfig is an arbitrary plugin configuration object.
+type pluginConfig interface{}
+
+// New returns a new config.Config
+func New() config.Config {
+	return &cfg{Version: Version}
+}
+
+func init() {
+	config.Register(Version, New)
+}
+
+// GetVersion implements config.Config
+func (c cfg) GetVersion() config.Version {
+	return c.Version
+}
+
+// GetDomain implements config.Config
+func (c cfg) GetDomain() string {
+	return c.Domain
+}
+
+// SetDomain implements config.Config
+func (c *cfg) SetDomain(domain string) error {
+	c.Domain = domain
+	return nil
+}
+
+// GetRepository implements config.Config
+func (c cfg) GetRepository() string {
+	return c.Repository
+}
+
+// SetRepository implements config.Config
+func (c *cfg) SetRepository(repository string) error {
+	c.Repository = repository
+	return nil
+}
+
+// GetName implements config.Config
+func (c cfg) GetName() string {
+	return c.Name
+}
+
+// SetName implements config.Config
+func (c *cfg) SetName(name string) error {
+	c.Name = name
+	return nil
+}
+
+// GetLayout implements config.Config
+func (c cfg) GetLayout() string {
+	return c.Layout
+}
+
+// SetLayout implements config.Config
+func (c *cfg) SetLayout(layout string) error {
+	c.Layout = layout
+	return nil
+}
+
+// IsMultiGroup implements config.Config
+func (c cfg) IsMultiGroup() bool {
+	return c.MultiGroup
+}
+
+// SetMultiGroup implements config.Config
+func (c *cfg) SetMultiGroup() error {
+	c.MultiGroup = true
+	return nil
+}
+
+// ClearMultiGroup implements config.Config
+func (c *cfg) ClearMultiGroup() error {
+	c.MultiGroup = false
+	return nil
+}
+
+// IsComponentConfig implements config.Config
+func (c cfg) IsComponentConfig() bool {
+	return c.ComponentConfig
+}
+
+// SetComponentConfig implements config.Config
+func (c *cfg) SetComponentConfig() error {
+	c.ComponentConfig = true
+	return nil
+}
+
+// ClearComponentConfig implements config.Config
+func (c *cfg) ClearComponentConfig() error {
+	c.ComponentConfig = false
+	return nil
+}
+
+// Resources implements config.Config
+func (c cfg) ResourcesLength() int {
+	return len(c.Resources)
+}
+
+// HasResource implements config.Config
+func (c cfg) HasResource(gvk resource.GVK) bool {
+	gvk.Domain = "" // Version 3 alpha does not include domain per resource
```

Comment:
Version 3 will, but this PR doesn't change v3-alpha, I was planning to create the v3 config in a separate PR as this is already pretty big.

Reply:
We need to fix it to get this PR merged. Ansible/Helm plugins in SDK are failing with this approach because their boilerplates uses `.Resource.Domain` which is returning nil.

Reply:
> We need to fix it to get this PR merged. Ansible/Helm plugins in SDK are failing with this approach because their boilerplates use `.Resource.Domain` which is returning nil.

This doesn't have to do with the config file in any sense. SDK is not setting the `Domain` field in the `Options` struct, and therefore the created `Resource` has no `Domain` set.